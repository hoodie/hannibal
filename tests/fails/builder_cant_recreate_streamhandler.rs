use futures::stream;
use minibal::{prelude::*, RestartableActor};

#[derive(Default)]
struct MyActor(&'static str);
impl Actor for MyActor {}

impl StreamHandler<i32> for MyActor {
    async fn handle(&mut self, _ctx: &mut Context<Self>, msg: i32) {
        println!("[Actor {}] Received: {}", self.0, msg);
    }
}
// TODO: can we encode the restart strategy in an associated type or as a trait function?
impl RestartableActor for MyActor {}

#[tokio::main]
async fn main() {
    let addr = minibal::build(MyActor("Caesar"))
        .unbounded()
        .recreate_from_default()
        .with_stream(stream::iter(vec![17, 19])) // this shouldn't work
        .spawn();
    addr.await
}
