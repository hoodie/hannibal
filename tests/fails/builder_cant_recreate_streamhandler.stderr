error[E0599]: no method named `with_stream` found for struct `ActorBuilderWithChannel<MyActor, hannibal::actor::restart_strategy::RecreateFromDefault>` in the current scope
  --> tests/fails/builder_cant_recreate_streamhandler.rs:21:10
   |
18 |       let addr = hannibal::build(MyActor("Caesar"))
   |  ________________-
19 | |         .unbounded()
20 | |         .recreate_from_default()
21 | |         .with_stream(stream::iter(vec![17, 19])) // this shouldn't work
   | |         -^^^^^^^^^^^ method not found in `ActorBuilderWithChannel<MyActor, hannibal::actor::restart_strategy::RecreateFromDefault>`
   | |_________|
   |
   |
   = note: the method was found for
           - `ActorBuilderWithChannel<A, hannibal::actor::restart_strategy::NonRestartable>`
